### PR TITLE
Adds support for showing the Pulumi Copilot sidebar on the docs site

### PR DESCRIPTION
**Pull request by foot**
### Proposed changes

Adds support for showing the Pulumi Copilot sidebar on the docs site

#### Notes

- Adds a route to the copilot host so that the iframe is SAMEORIGIN with the docs
  - This allows it to `React.portal` into the doc and interact w/ the page a bit.
  - This requires some tweaks to SecurityPolicy to allow the Copilot path to be iframe'd too

### CODE REVIEW

File: infrastructure/index.ts
Comment (foot) [inline infrastructure/index.ts:63]: revert this

File: infrastructure/index.ts
Comment (foot) [inline infrastructure/index.ts:63]: Revert before merging

**Review by cnunciato — APPROVED** [review]
Looks great @foot! Leaving an approval to unblock, but two things before we merge:

* I'd love to get a 👍🏻 from @sean1588 as well, just so we're all on the same page with this
* We'll need to fix up those lines in `infrastructure/index.ts` once the `prod` stack is ready to go

File: infrastructure/index.ts
Comment (sean1588) [inline infrastructure/index.ts:63]: can we make the stack reference itself a config value that is specified in the stack config, instead of hard coding it, since it appears to be environment specific?

unless we only have one environment for the ai stack that all docs stacks will reference then maybe this is ok

File: layouts/partials/copilot/sidebar.html
Comment (sean1588) [inline layouts/partials/copilot/sidebar.html:4]: just curious here, do we still want to show these clickable example prompts in the docs section of the site since they all seem cloud console specific to querying things about the org? I imagine this would need to be handled elsewhere though and not in this PR.

Screen Shot 2024-09-26 at 5 42 48 PM

**Review by sean1588 — APPROVED** [review]
These changes look good to me @foot! Nice work here!!

File: layouts/partials/copilot/sidebar.html
Comment (foot) [reply layouts/partials/copilot/sidebar.html:4]: Good one! Will make a follow up.

File: infrastructure/index.ts
Comment (foot) [reply infrastructure/index.ts:63]: We'll only have a one env really. Its a bit hard to align all the CORS etc to get this going on the test previews for now.

But I still like the idea a lot. Will do a follow up PR.

**Pull request merged**
